descartes_planner/src/sparse_planner.cpp

```diff
@@ -767,9 +767,11 @@ int SparsePlanner::interpolateSparseTrajectory(const SolutionArray& sparse_solut
           double dt = timing_cache_[pos].upper;
```

Comment:
```c++
const descartes_core::TimingConstraint& tm = timing_cache_[pos];
```
followed by

```c++
if (tm.isSpecified() && !robot_model->isValidMove(last_joint_pose, aprox_interp, tm.upper))
```

So that we're using the API for checking if the time is specified and not looking at the raw data. That's all.